ServerCore/Pages/Submissions/AuthorIndex.cshtml.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Submissions
+{
+    public class AuthorIndexModel : EventSpecificPageModel
+    {
+        private readonly ServerCore.DataModel.PuzzleServerContext _context;
+
+        public AuthorIndexModel(ServerCore.DataModel.PuzzleServerContext context)
+        {
+            _context = context;
+        }
+
+        public IList<Submission> Submissions { get; set; }
+                        
+        public async Task OnGetAsync(int? puzzleId)
+        {
+            if (puzzleId == null)
+            {
+                Submissions = await _context.Submissions.ToListAsync();
+            }
+            else
+            {
+                Submissions = await _context.Submissions.Where((s) => s.Puzzle != null && s.Puzzle.ID == puzzleId).ToListAsync();
```

Comment:
Should this be sorted (perhaps by submission time)?

Reply:
Aren't these automatically sorted by time since they get added in order?